iceoryx_posh/test/moduletests/test_popo_sample.cpp

```diff
@@ -14,19 +15,24 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#include "iceoryx_posh/popo/publisher.hpp"
 #include "iceoryx_posh/popo/sample.hpp"
-#include "iceoryx_posh/popo/typed_publisher.hpp"
 #include "iceoryx_utils/cxx/unique_ptr.hpp"
 #include "test.hpp"
 
 using namespace ::testing;
 using ::testing::_;
 
+// anonymous namespace to prevent linker issues or sanitizer false positives
+// if a struct with the same name is used in other tests
+namespace
+{
 struct DummyData
 {
     DummyData() = default;
-    int val = 42;
+    uint32_t val = 42;
```

Comment:
:octopus:  <- @dkroenke

Reply:
btw. that was the reason for the address sanitizer issue. There were 4 structs with the same name, three with uint64_t and one with int all in the same binary. Somehow the address sanitizer mixed them up.